sdk/documenttranslation/azure-ai-documenttranslation/tests/testcase.py

```diff
@@ -19,4 +50,70 @@ class DocumentTranslationTest(AzureTestCase):
         self.vcr.match_on = ["path", "method", "query"]
+        self.recording_processors.append(OperationLocationReplacer())
+        self.storage_name = os.getenv("DOCUMENTTRANSLATION_STORAGE_NAME", "redacted")
+        self.storage_endpoint = "https://" + self.storage_name + ".blob.core.windows.net/"
+        self.storage_key = os.getenv("DOCUMENTTRANSLATION_STORAGE_KEY")
+        self.scrubber.register_name_pair(
+            self.storage_endpoint, "https://redacted.blob.core.windows.net/"
+        )
+        self.scrubber.register_name_pair(
+            self.storage_name, "redacted"
+        )
+        self.scrubber.register_name_pair(
+            self.storage_key, "fakeZmFrZV9hY29jdW50X2tleQ=="
+        )
+
+    def _setup(self, data=None, blob_prefix=""):
+        """Creates a source and target container.
+
+        Pass data in as bytes (or as a list[bytes] to create more than one blob) in the source container.
+        """
+        if self.is_live:
+            self.source_container_sas_url = self.create_source_container(
+                data=data or b'This is written in english.',
+                blob_prefix=blob_prefix
+            )
+            self.target_container_sas_url = self.create_target_container()
+        else:
+            self.source_container_sas_url = "source_container_sas_url"
+            self.target_container_sas_url = "target_container_sas_url"
+
+    def create_source_container(self, data, blob_prefix=""):
+        container_name = "src" + str(uuid.uuid4())
+        container_client = ContainerClient(self.storage_endpoint, container_name,
+                                           self.storage_key)
+        container_client.create_container()
+        if isinstance(data, list):
+            for blob in data:
+                container_client.upload_blob(name=blob_prefix+str(uuid.uuid4()) + ".txt", data=blob)
+        else:
+            container_client.upload_blob(name=blob_prefix+str(uuid.uuid4())+".txt", data=data)
+        return self.generate_sas_url(container_name, "rl")
+
+    def create_target_container(self):
+        container_name = "target" + str(uuid.uuid4())
+        container_client = ContainerClient(self.storage_endpoint, container_name,
+                                           self.storage_key)
+        container_client.create_container()
+
+        return self.generate_sas_url(container_name, "racwdl")
+
+    def generate_sas_url(self, container_name, permission):
+
+        sas_token = generate_container_sas(
+            self.storage_name,
+            container_name,
+            account_key=self.storage_key,
+            permission=permission,
+            expiry=datetime.datetime.utcnow() + datetime.timedelta(hours=2)
+        )
+        container_sas_url = self.storage_endpoint + container_name + "?" + sas_token
+        self.scrubber.register_name_pair(
+            sas_token, "container_sas_url"
+        )
+        return container_sas_url
```

Comment:
SAS urls are difficult to scrub especially when you have expiry added. Try using [this](https://github.com/Azure/azure-sdk-for-python/blob/master/tools/azure-sdk-tools/devtools_testutils/azure_testcase.py#L357-L370) method in `AzureTestCase` here.

Reply:
thanks, will update